### ISSUE TITLE
Issues with xgb.train and multi:softprob

### ISSUE THREAD

**Issue by DeluxeAnalyst** [dmlc/xgboost#745]
I am getting an issue with trying to use xgb.train with early stopping round, and multi:softprob as the objective. 

The error I am getting says my class is set to 12, but its expecting 12, which is exactly what I passed it. I dont know if there is an issue with having the label input as part of the DMatrix?

Here is exact error:

Error in xgb.iter.update(bst$handle, dtrain, i - 1, obj) : 
  SoftmaxMultiClassObj: label must be in [0, num_class), num_class=12 but found 12 in label

Here is my sample code:
dval <- xgb.DMatrix(data=data.matrix(val[,-1]),label = y_val)
dtrain <- xgb.DMatrix(data.matrix(train[,-1]),label = y)
watchlist = list(val=dval,train=dtrain)

bst <- xgb.train(data = xgb.DMatrix(data.matrix(train[,-1]),label = y),
                     max.depth=3, 
                     nround=1000,
                     eval_metric = "merror",
                     objective = "multi:softprob",
                     num_class = 12,
                     early.stop.round = 30,
                     watchlist = list(val=dval,train=dtrain),
                     maximize = FALSE,
                     verbose=0)

**Comment by DeluxeAnalyst** [comment]
I dont know why this fixed it, but I wrapped my Y variable in a as.numeric(as.character(y)) and it looks like its working now.

**Comment by tqchen** [comment]
The reason is the label need to be in [0, 12) (not including 12), but can contain 0
